stl/inc/mdspan

```diff
@@ -935,14 +935,15 @@ public:
     constexpr mdspan& operator=(const mdspan&) = default;
     constexpr mdspan& operator=(mdspan&&)      = default;
 
-    // TRANSITION, P2128R6 (Multidimensional subscript operator)
+#ifdef __clang__ // TRANSITION, P2128R6

```

Comment:
Should we consider testing `__cpp_multidimensional_subscript` instead? Writing "light-up" code has advantages and disadvantages:

* We properly react to newer compilers instead of having to change the STL again
* However, because the compiler can activate new STL codepaths without our explicit blessing, this can result in essentially untested configurations being immediately experienced by users
  + However x2, since MSVC compiler development and EDG updates run the STL's test suite, this would prevent broken compiler support from shipping.

I think this argues in favor of using the feature-test macro.